api_app/analyzers_manager/file_analyzers/debloat.py

```diff
@@ -0,0 +1,53 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+import logging
+import os
+from base64 import b64encode
+
+import debloat
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+
+logger = logging.getLogger(__name__)
+
+
+class Debloat(FileAnalyzer):
+    def run(self):
+        file_path = self.filepath  # Get the file path
+        output_file = f"{file_path}.debloated"
+
+        result = dict()
+        try:
+            logger.info(f"Running Debloat on {file_path}")
+            binary = self.read_file_bytes()
+            file_size = os.path.getsize(file_path)
+
+            debloat.processor.process_pe(
+                binary,
+                out_path=str(output_file),
+                last_ditch_processing=True,
+                cert_preservation=True,
+                log_message=print,
+                beginning_file_size=file_size,
+            )
+
+            if not os.path.exists(output_file) or os.path.isdir(output_file):
+                result["error"] = "Debloat failed to produce output"
```

Comment:
you need to raise AnalyzerRunException and populate self.report.errors. Please see other analyzers